cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/trigger/TriggerInfos.java

```diff
@@ -0,0 +1,92 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.schedule.trigger;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.cdap.api.schedule.Trigger;
+import io.cdap.cdap.api.schedule.TriggerInfo;
+import io.cdap.cdap.api.schedule.TriggeringScheduleInfo;
+import io.cdap.cdap.api.workflow.WorkflowToken;
+import io.cdap.cdap.common.app.RunIds;
+import io.cdap.cdap.internal.app.runtime.ProgramOptionConstants;
+import io.cdap.cdap.internal.app.runtime.schedule.DefaultTriggeringScheduleInfo;
+import io.cdap.cdap.internal.app.runtime.workflow.WorkflowTokenCodec;
+import io.cdap.cdap.spi.events.trigger.mapping.TriggeringPropertyMapping;
+import org.apache.twill.api.RunId;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Helper functions for {@link TriggerInfo}
+ */
+public class TriggerInfos {
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(RunId.class, new RunIds.RunIdCodec())
+    .registerTypeAdapter(TriggerInfo.class, new TriggerInfoCodec())
+    .registerTypeAdapter(WorkflowToken.class, new WorkflowTokenCodec())
+    .create();
+  public static final String TRIGGERING_PROPERTIES_MAPPING = "triggering.properties.mapping";
+
+  /**
+   * Deserialize {@link TriggeringScheduleInfo} if SystemArgs contains the key 'triggeringScheduleInfo'
+   * else returns null
+   * @param sysArgs
+   * @return {@link TriggeringScheduleInfo}
+   */
+  @Nullable
+  public static TriggeringScheduleInfo deserializeTriggeringScheduleInfo(Map<String, String> sysArgs) {
+    if (!sysArgs.containsKey(ProgramOptionConstants.TRIGGERING_SCHEDULE_INFO)) {
+      return null;
+    }
+    // since only implementation we use is `DefaultTriggeringScheduleInfo`
+    return GSON.fromJson(
+      sysArgs.get(ProgramOptionConstants.TRIGGERING_SCHEDULE_INFO),
+      DefaultTriggeringScheduleInfo.class);
+  }
+
+  /**
+   * Deserialize {@link Trigger.Type} if SystemArgs contains the key 'triggeringScheduleInfoType'
+   * else returns null
+   * @param sysArgs
+   * @return {@link Trigger.Type}
+   */
+  @Nullable
+  public static Trigger.Type deserializeTriggerType(Map<String, String> sysArgs) {
+    if (!sysArgs.containsKey(ProgramOptionConstants.TRIGGERING_SCHEDULE_INFO_TYPE)) {
+      return null;
+    }
+    return GSON.fromJson(sysArgs.get(ProgramOptionConstants.TRIGGERING_SCHEDULE_INFO_TYPE), Trigger.Type.class);
+  }
+
+  /**
+   * Deserializes {@link TriggeringPropertyMapping} if properties contain the key 'triggering.properties.mapping'
+   * else returns null
+   * @param triggeringScheduleInfo
+   * @return
+   */
+  @Nullable
+  public static TriggeringPropertyMapping deserializeTriggeringPropertyMapping(
+    TriggeringScheduleInfo triggeringScheduleInfo) {
+    if (!triggeringScheduleInfo.getProperties().containsKey(TRIGGERING_PROPERTIES_MAPPING)) {
```

Comment:
it looks like the mapping is already there in the schedule info properties. I think we should just keep it there instead of handling it specially as a separate field. If an event writer wants to use it, they will have to have this logic that looks at the schedule properties to find the mapping.

This is because this property mapping is something that only the pipeline app knows about. It is not something the platform layer should know anything about.

If we really need to, we could add some module at the cdap-etl-* level that has TriggeringPropertyMapping and relevant classes. An event writer that needs it could add that module as a dependency and use it. But I think it's better to just leave it out completely for now and add it at a later date if needed.